### ISSUE TITLE
[Feature] useCookie

### ISSUE THREAD

**Issue by AlexandreAWE** [streamich/react-use#594]
Hello guys,

What do you think about a useCookie hook ?

```js
const [foo, setFoo, deleteFoo] = useCookie('foo');

setFoo('value', 365); // setFoo(value:string, expirationDays:number) => void
deleteFoo()
```

Possible implentation : 
```typescript
export function useCookie(cookieName:string): [string, (value: string, days?: number) => void, () => void] {
  const [cookieValue, setCookieValue] = useState(() => {
    const v = document.cookie.match('(^|;) ?' + cookieName + '=([^;]*)(;|$)');
    return v ? v[2] : null;
  });

  const updateCookie = (value: string, days: number = 365) => {
    const d = new Date();
    d.setTime(d.getTime() + 24 * 60 * 60 * 1000 * days);
    document.cookie = cookieName + '=' + value + ';path=/;expires=' + d.toUTCString();
    setCookieValue(value);
  };

  const deleteCookie = () => {
    updateCookie('', -1);
    setCookieValue(null);
  };

  return [cookieValue, updateCookie, deleteCookie];
}
```

Inspired from : https://plainjs.com/javascript/utilities/set-cookie-get-cookie-and-delete-cookie-5/

Waiting for reviews.

Thanks

**Comment by CloudPower97** [comment]
Hey @AlexandreAWE , I was about to open an issue myself with the same exact request.
I think this would be really useful.

@AlexandreAWE We might want to use [`js-cookie`](https://github.com/js-cookie/js-cookie) to handle this.

@streamich Let us know your thought about this :smile:

**Comment by streamich** [comment]
Would be interesting also to listen and re-render on cookie changes, [see](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged).

**Comment by streamich** [comment]
- [ ] return latest value of cookie
- [ ] function to set cookie
- [ ] function to delete cookie, possibly by ussing set function, and setting cookie to `null`
- [ ] listen for cookie changes and re-render
- [ ] some sensible mock of this hook for server-side rendering

**Comment by ayush987goyal** [comment]
@AlexandreAWE @CloudPower97 @streamich Hey guys! Could you please review the above PR?

**Comment by ayush987goyal** [comment]
@streamich  This issue can be closed. Merged with #878
